### PR TITLE
Fix error message when successively loading the same dataset

### PR DESCRIPTION
**Pull request by brownj85**
**Context:**
Attempting to partially load a dataset after fully loading results in an error:
```
>>> H2datasets = qml.data.load("qchem", molname="H2", basis="STO-3G", bondlength=1.1)
>>> part = qml.data.load("qchem", molname="H2", basis="STO-3G", bondlength=1.1, attributes=["molecule", "fci_energy"])[0]
OSError: Unable to open file (file is already open for read-only)
```

However, fuly loading twice in a row does not result in an error:
```
>>> H2datasets = qml.data.load("qchem", molname="H2", basis="STO-3G", bondlength=1.1)
>>> H2dataset = qml.data.load("qchem", molname="H2", basis="STO-3G", bondlength=1.1)[0]
```

This happens because `_download_partial` needs to open the local dataset file in read-write mode, so the attributes can be copied in. But `load()` opens the local files in read-only mode. The conflict occurs because two HDF5 file objects cannot open the same file with different read-write modes.

**Description of the Change:**
`load()` now opens the local dataset files in read-write (`'a'`) mode

**Benefits:**
Eliminates unexpected and confusing error.

**Possible Drawbacks:**
Users may not realize they are altering their local copy of the dataset, if they make modifications.

**Related GitHub Issues:**

### CODE REVIEW

**Review by DSGuala — APPROVED** [review]
Gave it a quick test and it seems to solve the problem! :+1:

File: pennylane/data/base/dataset.py [hunk 1]
```diff
@@ -223,13 +223,11 @@ def open(
         Returns:
             Dataset object from file
         """
-        filepath = Path(filepath).expanduser()
```
Comment (timmysilv) [inline pennylane/data/base/dataset.py:226]: is the expanduser no longer needed?

Comment (brownj85) [reply pennylane/data/base/dataset.py:226]: nope, it definitely is - good catch

**Review by timmysilv — APPROVED** [review]
nice and simple

**Pull request merged**